src/Maestro/Maestro.Web/Startup.cs

```diff
@@ -392,6 +403,52 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env)
                 app.UseHttpsRedirection();
             }
 
+            // Add security headers
+            app.Use(
+                (ctx, next) =>
+                {
+                    ctx.Response.OnStarting(() =>
+                    {
+                        if (!ctx.Response.Headers.ContainsKey("X-XSS-Protection"))
+                        {
+                            ctx.Response.Headers.Add("X-XSS-Protection", "1");
+                        }
+
+                        if (!ctx.Response.Headers.ContainsKey("X-Frame-Options"))
+                        {
+                            ctx.Response.Headers.Add("X-Frame-Options", "DENY");
+                        }
+
+                        if (!ctx.Response.Headers.ContainsKey("X-Content-Type-Options"))
+                        {
+                            ctx.Response.Headers.Add("X-Content-Type-Options", "nosniff");
+                        }
+
+                        if (!ctx.Response.Headers.ContainsKey("Referrer-Policy"))
+                        {
+                            ctx.Response.Headers.Add("Referrer-Policy", "no-referrer-when-downgrade");
+                        }
+
+                        if (!ctx.Response.Headers.ContainsKey("Cache-Control"))
+                        {
+                            ctx.Response.Headers.Add("Cache-Control", "no-cache, no-store, must-revalidate, proxy-revalidate");
+                        }
+                        else
+                        {
+                            ctx.Response.Headers.Append("Cache-Control", ", no-store, must-revalidate, proxy-revalidate");
```

Comment:
Why doesn't this one contain "no-cache"? "no-store", and will, to my knowledge, on all browsers except some ancient IE's, override "no-cache"... But it's odd the lists aren't the same.

I also think it's safe to just always call "Add"... having 2 headers with the same name should concatenate all the command values together... based on the spec... so we shouldn't need the original if...

Reply:
It doesn't contain no-cache because that's what gets added by some part of the middleware when the header is already present. I can try adding it and see if it can handle the duplicate but I think it wasn't happy when I tried that (I'll double check, I've done a lot of iterations so I'm not entirely sure).

Add throws an exception if the header already exists. If the spec doesn't call for that then I don't know what layer is doing it, but something in our stack isn't willing to concatenate.